cmd/erasure-decode.go

```diff
@@ -289,12 +289,15 @@ func (e Erasure) Decode(ctx context.Context, writer io.Writer, readers []io.Read
 }
 
 // Heal reads from readers, reconstruct shards and writes the data to the writers.
-func (e Erasure) Heal(ctx context.Context, writers []io.Writer, readers []io.ReaderAt, totalLength int64) (derr error) {
+func (e Erasure) Heal(ctx context.Context, writers []io.Writer, readers []io.ReaderAt, totalLength int64, prefer []bool) (derr error) {
 	if len(writers) != e.parityBlocks+e.dataBlocks {
 		return errInvalidArgument
 	}
 
 	reader := newParallelReader(readers, e, 0, totalLength)
+	if len(readers) == len(prefer) {
```

Comment:
There is always true. That's on purpose?
```go
readers := make([]io.ReaderAt, len(latestDisks))
prefer := make([]bool, len(latestDisks))
```

Reply:
yes it checks if the input `prefer` is non-nil and it matches number of readers.  @jiuker

Reply:
@harshavardhana I mean We should check the `prefer`'s `true` length?

Reply:
no not really @jiuker it must be same as available readers if its not then we do not enable preferential reading.